File: packages/sdk-js/core/interop/lite_mode/python/token.py
"""
PEAC Protocol v0.9.1
Lite Mode Token - Python
Apache-2.0 License
"""
import base64
import time

_issued_tokens = set()

def generate_token(agent_id):
    token = base64.b64encode(f"{agent_id}:{int(time.time())}".encode()).decode()
    _issued_tokens.add(token)
    return token

def validate_token(token):
    return token in _issued_tokens

File: archived/core/interop/lite_mode/node/token.js
/**
 * PEAC Protocol v0.9.1
 * Lite Mode Token - Node.js
 * Apache-2.0 License
 */
const issuedTokens = new Set();
function generateToken(agentId) {
  const token = Buffer.from(`${agentId}:${Date.now()}`).toString('base64');
  issuedTokens.add(token);
  return token;
}
function validateToken(token) {
  return issuedTokens.has(token);
}
module.exports = { generateToken, validateToken };
